### ISSUE TITLE
Confusing behavior when collapsing/expanding comment threads if several exist on the same line

### ISSUE THREAD

**Issue by IlyaBiryukov** [microsoft/vscode#77155]
- VSCode Version: 1.36.0
- OS Version: Windows 10

**Repro steps:**
1. With Live Share extension 1.0.488, start a collaboration session
2. Get two comment threads on a single line.
     * Comment on a line, 3 for example
     * Comment on a second line, 4 for example
     * Delete line 3

   Now there are two comment threads on line 3 (the line that moved up from 4)
3. Expand/Collapse the comments with the marker 

Collapsing/opening such comments might need a rework. Currently I can collapse each individually, and if all are collapsed (or all open), the box on the gray line opens (or closes) all threads. However, if only some of the threads are collapsed, the gray box toggles between the set of open threads and the threads which were collapsed. It never collapses all threads to return to code, and requires each remaining thread to be collapsed individually.

Originally reported in Live Share issue
https://github.com/MicrosoftDocs/live-share/issues/2472

**Comment by hermannloose** [comment]
As I was just about to file the same thing, here is the screencast I created for illustration:

![confusing-expand-collapse](https://user-images.githubusercontent.com/163766/151003879-446c5df1-3e60-45e2-8535-60b3bcdb03ca.gif)

**Comment by laurentlb** [comment]
I agree the current behavior is a bit confusing. Clicking in the gutter is a common way to collapse open comments. However, the workflow fails when multiple comments are on the same line. This forces the user to think and use the alternative workflow (top-right icon).

Suggested behavior:
- if at least one comment is expanded, collapse all comments on the line
- otherwise, expand all the comments

The condition could also be reversed (expand if the line has both expanded and collapsed comments).

**Comment by hermannloose** [comment]
Since we just came across the relevant code again in another context: clicks on the gutter icon currently toggle the collapsible states for all threads on that line independently here: https://github.com/microsoft/vscode/blob/2552748fc6544b6f4f05fb224858e6646ca74a9c/src/vs/workbench/contrib/comments/browser/commentsController.ts#L769

`ReviewZoneWidget._isExpanded` already exists as a private property, if there was a public getter for it, behavior like the one suggested by @laurentlb above could probably be implemented very locally in the location I linked to above.

@alexr00 let us know if that sounds reasonable and we can make a PR.

**Comment by alexr00** [comment]
@hermannloose yes, this sounds reasonable to me. l would accept a PR!

**Comment by hermannloose** [comment]
@alexr00 I opened #176967.
